raft/src/main/java/org/apache/kafka/raft/ReplicatedLog.java

```diff
@@ -280,11 +280,19 @@ default long truncateToEndOffset(OffsetAndEpoch endOffset) {
     Optional<RawSnapshotReader> readSnapshot(OffsetAndEpoch snapshotId);
 
     /**
-     * Returns the latest snapshot id if one exists.
+     * Returns the latest readable snapshot if one exists.
      *
-     * @return an Optional snapshot id of the latest snashot if one exists, otherwise returns an
-     * empty Optional
+     * @return an Optional with the latest readable snapshot, if one exists, otherwise
+     *         returns an empty Optional
      */
+    Optional<RawSnapshotReader> latestSnapshot();
```

Comment:
would "openLatestSnapshot" be a better name, given that the snapshot reader will need to be closed later?

Reply:
`RawSnapshotReader` are not `Closeable` or `AutoCloseable`. The specific implementation `FileRawSnapshotReader` is `AutoCloseable`.

This was done to match Kafka ownership model. For example, kraft sends the `FileChannel` owned by `RawSnapshotReader` to the networking layer when replying to `FetchSnapshot` requests.

`FileRawSnapshotReader` are closed as part of the clean up process. 60 seconds (if I remember correctly) after `maybeClean` has determine that they should be deleted.